types/es6template/index.d.ts

```diff
@@ -0,0 +1,18 @@
+// Type definitions for es6template 1.0
+// Project: https://github.com/zalmoxisus/es6-template
+// Definitions by: Nathan Bierema <https://github.com/Methuselah96>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+interface Default {
+    render: (str: string, locals: Record<string, unknown>) => string;
+
+    // This rule is disabled because the caller of `compile()` knows what the type of the `locals` parameter should be when
+    // calling the `compile` function.
+    // tslint:disable-next-line:no-unnecessary-generics
+    compile: <Locals>(str: string) => ((locals: Locals) => string);
```

Comment:
Given that the user still would need to cast or provide the type for Locals, it shouldnt be generic.
```suggestion
    compile: (str: string) => ((locals: object) => string);
```

Reply:
@sheetalkamat Thanks for reviewing this! The point is that they should be forced to provide the type for Locals in order to have type safety.